compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -126,36 +128,14 @@ pub fn report_unstable(
 /// Checks whether an item marked with `deprecated(since="X")` is currently
 /// deprecated (i.e., whether X is not greater than the current rustc version).
 pub fn deprecation_in_effect(depr: &Deprecation) -> bool {
```

Comment:
Can this be made an inherent method on Deprecation or DeprecatedSince?

Reply:
Done; added `pub fn is_in_effect(&self) -> bool` on `Deprecation`.